src/raw/http_util/resolver.rs

```diff
@@ -0,0 +1,69 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use hyper::client::connect::dns::Name;
+use once_cell::sync::OnceCell;
+use reqwest::dns::{Addrs, Resolve};
+use trust_dns_resolver::TokioAsyncResolver;
+
+/// a global resolver to reuse dns cache
+///
+/// # Note:
+/// this structure contains nothing, only used as an entry point
+/// to call from `trust_dns_resolver`
+#[derive(Debug)]
+pub struct DnsClient {}
```

Comment:
Oh, I don't like this API design. I prefer to put `TokioAsyncResolver` inside `DnsClient` and provide a global for `OnceCell<Arc<DnsClient>>` instead.

In this way, our user can decide how to use them.